pytest_splunk_addon_ui_smartx/base_test.py

```diff
@@ -63,246 +54,92 @@ def __init__(
         self.splunk_mgmt_url = splunk_mgmt_url
         self.cred = cred
         self.test_case = test_case
-        self.skip_saucelab_job = False
 
         selenium_host = os.environ.get("SELENIUM_HOST")
 
-        if "grid" in browser:
-            self.skip_saucelab_job = True
-            debug = True
-        elif selenium_host:
-            self.skip_saucelab_job = True
-        if debug or selenium_host:
-            pass
-        else:
-            self.init_sauce_env_variables()
-
         try:
             if browser == "firefox":
-                if debug:
+                if local_run:
                     self.browser = webdriver.Firefox(
-                        firefox_options=self.get_local_firefox_opts(headless),
-                        log_path="selenium.log",
+                        firefox_options=SeleniumHelper.get_local_firefox_opts(headless)
                     )
                 elif selenium_host:
                     self.browser = webdriver.Remote(
                         command_executor=f"{selenium_host}:4444/wd/hub",
-                        options=self.get_local_firefox_opts(headless_run=False),
+                        options=SeleniumHelper.get_local_firefox_opts(
+                            headless_run=False
+                        ),
                     )
                     self.browser.implicitly_wait(3)
                 else:
-                    self.browser = webdriver.Remote(
-                        command_executor="https://ondemand.saucelabs.com:443/wd/hub",
-                        desired_capabilities=self.get_sauce_firefox_opts(
-                            browser_version
-                        ),
+                    raise Exception(
+                        f"Firefox tests have to be run either with --local or in CI environment with selenium host!"
                     )
+
             elif browser == "chrome":
-                if debug:
+                if local_run:
                     self.browser = webdriver.Chrome(
-                        chrome_options=self.get_local_chrome_opts(headless),
-                        service_args=["--verbose", "--log-path=selenium.log"],
```

Comment:
What's the reason to remove this file?

Reply:
selenium.log file is not being used anywhere as far as I know. Logging improvements could be part of broader refactor in the future, together with defining proper location for selenium logs.